src/lib/types.hpp

```diff
@@ -180,6 +180,11 @@ enum class PredicateCondition {
   IsNotNull
 };
 
+struct JoinPredicate {
```

Comment:
This looks remarkably similar to OperatorJoinPredicate, it should be possible to just use the latter.